stl/inc/chrono

```diff
@@ -2086,6 +2090,721 @@ namespace chrono {
+            {sys_seconds{seconds{1483228800}}, true, seconds{26}},
+        };
+        constexpr size_t _Pre_2018_count = 27;
+
+        size_t _New_reg_ls_size;
+        unique_ptr<_RegistryLeapSecondInfo[], decltype(&__std_decalloc_reg_leap_seconds)> _Reg_ls_data(
+            __std_tzdb_get_reg_leap_seconds(
+                (_STD max)(_Current_size, _STD size(_Known_leap_seconds)) - _Pre_2018_count, &_New_reg_ls_size),
+            &__std_decalloc_reg_leap_seconds);
+
+        const size_t _New_size = _Pre_2018_count + _New_reg_ls_size;
+        if (_New_reg_ls_size > _Current_size && !_Reg_ls_data) {
+            _Xbad_alloc();
+        } else if (_New_reg_ls_size == 0 && _Reg_ls_data) {
+            _XGetLastError();
+        }
+
+        decltype(tzdb::leap_seconds) _Leap_sec_info;
+        bool _All_ls_positive = true;
+
+        if (_New_size > _Current_size) {
+            _Leap_sec_info.reserve(_New_size);
+            _Leap_sec_info.insert(
+                _Leap_sec_info.begin(), _STD cbegin(_Known_leap_seconds), _STD cend(_Known_leap_seconds));
+
+            for (size_t _Idx = 0; _Idx < _New_reg_ls_size; ++_Idx) {
+                // Leap second happens at ls._Hour:59:59. We store the next second after, so add a whole hour.
+                const auto& _Ls = _Reg_ls_data[_Idx];
+                const auto _Date =
+                    static_cast<sys_days>(year_month_day{year{_Ls._Year}, month{_Ls._Month}, day{_Ls._Day}})
+                    + hours{_Ls._Hour + 1};
+                _Leap_sec_info.emplace_back(_Date, !_Ls._Negative, _Leap_sec_info.back()._Elapsed());
+                _All_ls_positive = _All_ls_positive && !_Ls._Negative;
+            }
+        }
+
+        return {_Leap_sec_info, _All_ls_positive};
+    }
+
+    // TRANSITION: work in progress
+    class tzdb_list {
+    private:
+        using _ListType = forward_list<tzdb, _Crt_allocator<tzdb>>;
+
+    public:
+        using const_iterator = _ListType::const_iterator;
+
+        tzdb_list(const tzdb_list&) = delete;
+        tzdb_list& operator=(const tzdb_list&) = delete;
+
+        tzdb_list() {
+            auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(0);
+            _Tzdb_list.emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});
+        }
+
+        _NODISCARD const tzdb& front() const noexcept {
+            _Shared_lock _Lk(_Tzdb_mutex);
+            return _Tzdb_list.front();
+        }
+
+        template <class... _ArgsTy>
+        void _Emplace_front(_ArgsTy&&... _Args) {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            _Tzdb_list.emplace_front(_STD forward<_ArgsTy>(_Args)...);
+        }
+
+        const tzdb& _Reload() {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(_Tzdb_list.front().leap_seconds.size());
+            if (!_Leap_sec.empty()) {
+                _Emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});

```

Comment:
I think you forgot about my previous comment about making this call `_Tzdb_list.emplace_front`